docs: add Netlify deploy process doc (#4650)
liferay/clay: docs/netlify_deploy.md

# Netlify deploy process

> We use Netlify to deploy clayui.com and a version of Storybook, the deployment process is very simple, the build happens in the service itself so we don't have a Github Actions configuration.

Reviewer: ```suggestion
We use [Netlify](https://www.netlify.com/) to deploy [clayui.com](https://clayui.com) and a version of Storybook. The deployment process is very simple, and the build happens in the service itself so we don't have a Github Actions configuration.
```

> A new version of clayui.com and storybook are triggered in different scenarios.

Reviewer: ```suggestion
A new version of clayui.com and storybook are generated in different scenarios:
```

-   **Deploy Previews** - every new PR a new build is made to show the changes made
-   **next.clayui.com** - Represents changes that are in the master branch
-   **next.storybook.clayui.com** - Represents changes that are in the master branch
-   **clayui.com** - Represents changes that are in the stable branch
-   **storybook.clayui.com** - Represents changes that are in the stable branch

## Deploy Previews

> All PR triggers a new build for clayui.com and storybook and generates a preview URL that allows you to preview the changes without having to download the PR locally and test.

Reviewer: ```suggestion
All pull requests trigger a new build of clayui.com and storybook and generate a preview URL that allows you to view the changes without having to download the changes locally.
```

PR could mean "public relations" or something else, I think it's better to be specific and use complete words.

> Netlify uses Github Checks to demonstrate the build status, when the build is being done it is possible to click `Details` on the corresponding check and check the build status as well as the logs. When done clicking on `Details` again will redirect to the corresponding preview that was generated.

Reviewer: ```suggestion
Netlify uses [Github Checks](https://docs.github.com/en/rest/reference/checks) to show information about the build status. During the build process, it's possible to click on `Details` to view the build output as well as the logs. Clicking on `Details` a second time will redirect to the corresponding preview that was generated.
```

> No need to be a member to view build status or preview

> <img width="916" alt="Screen Shot 2022-02-08 at 2 09 11 PM" src="https://user-images.githubusercontent.com/13750819/153067419-b0dad542-9d4b-4bdc-9378-aa7100d5a230.png">

Author: Oh yeah, good catch, this was automatically generated by GitHub when I added the image.

[Learn more about Deploy Previews to Netlify](https://docs.netlify.com/site-deploys/deploy-previews/)

## Configuration

> Netlify configuration for clayui.com and storybook is very simple, we don't need extra configuration for these cases or Github Actions for example, much of the configuration is done the first time we integrate with Netlify and connect the account with the Github repository.

Reviewer: I think we're repeating ourselves a bit here, and since you added a link to the Netlify's preview documentation, I think this isn't needed.

Author: Hmm yeah, you're right, I ended up talking about this in some other parts, I'll remove that.

> Netlify will observe all changes in the repository and trigger builds when necessary, in the above cases for example. It has its own build system, so we need to tell which command to run and it will do everything else.

Reviewer: ```suggestion
Netlify will observe all changes in the repository and trigger builds when necessary. We just need to specify which command to run and it will do  the rest.
```
